### PR TITLE
Allow disabling SystemChrome.setApplicationSwitcherDescription for Title Widget

### LINKED ISSUE #108259: SystemChrome.setApplicationSwitcherDescription makes the app NAME disappear in the application switcher
## Steps to Reproduce

MaterialApp-->WidgetsApp-->Title-->[SystemChrome.setApplicationSwitcherDescription](https://github.com/flutter/flutter/blob/bbdf617034/packages/flutter/lib/src/services/system_chrome.dart#L393)

image

Our app is a mix of Flutter and native development, flutter is only used for part of the functional UI。
SystemChrome.setApplicationSwitcherDescription makes the app NAME disappear in the application switcher！！！
image

It is hoped that this feature can be turned on or off by variable switch,
 for example [SystemChrome.setSystemUIOverlayStyle](https://github.com/flutter/flutter/blob/f852c46c877bc321c1705403cfb2b94e5976de3f/packages/flutter/lib/src/rendering/view.dart#L125)

image

```
➜  ~ flutter doctor -v
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-x64, locale
    zh-Hans-CN)
    • Flutter version 2.5.3 at /Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (9 months ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/junhualin/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS
      development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 103.0.5060.114

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by JunhuaLin**
Allow disabling SystemChrome.setApplicationSwitcherDescription for Title Widget

fix https://github.com/flutter/flutter/issues/108259

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by JunhuaLin** [comment]
Could you take a look? @danagbemava-nc

**Comment by JunhuaLin** [comment]
Is anybody paying attention here？

**Comment by Hixie** [comment]
It's probably more idiomatic to implement this by having WidgetsApp and MaterialApp take a builder method for the `Title` widget (which gets passed a `child`), and if you want to disable it you would just return `child` directly from that callback.

**Comment by Hixie** [comment]
(Not sure why this never got reviewed. I'll investigate. Sorry about the delay! Thanks for your contribution!)

File: packages/flutter/lib/src/widgets/title.dart [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Title extends StatelessWidget {
   Title({
     super.key,
     this.title = '',
+    this.automaticAppSwitcherDescAdjustment = true,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/title.dart:21]: The whole purpose of the Title widget is to populate the ApplicationSwitcher with the title. If you don't want that, you might as well just not use the Title widget.

**Comment by Hixie** [comment]
@JunhuaLin Thanks for your contribution. Will you be able to continue working on this? It's totally fine if not; we can take it over (though it might take longer). I'm just asking because I'm trying to clean out our review queue.

**Comment by Hixie** [comment]
Closing per comments above.

**Comment by Jackfinal** [comment]
好像你这个没有合并进去，请问一下，现在是怎么解决的

**Pull request closed without merging**